sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -330,4 +335,31 @@ class SQLConfSuite extends QueryTest with SharedSparkSession {
     }
     assert(e.getMessage.contains(config))
   }
+
+  test("log deprecation warnings") {
+    val logAppender = new AppenderSkeleton {
```

Comment:
nit: The same class `logAppender` seems to be defined in some places below, so can we define a helper method for this test purpose somewhere (e.g., `TestUtils`)?
```
$grep -nr "extends AppenderSkeleton" .
./catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveHintsSuite.scala:36:  class MockAppender extends AppenderSkeleton {
./catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala:525:    class MockAppender extends AppenderSkeleton {
./catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizerLoggingSuite.scala:42:  class MockAppender extends AppenderSkeleton {
./core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala:1766:    class TestAppender extends AppenderSkeleton {
./core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala:41:  class MockAppender extends AppenderSkeleton {
```

Reply:
it is slightly orthogonal to the PR but if you think it makes sense I will do that here.

Reply:
Yea, I think its ok in follow-up.